data/data-pipeline/data_pipeline/utils.py

```diff
@@ -198,6 +199,21 @@ def temp_folder_cleanup() -> None:
     remove_all_from_dir(data_path / "tmp")
 
 
+def check_first_run() -> bool:
+    """Checks if a local semaphore has been set to only run once the
+    calling function"""
+
+    data_path = settings.APP_ROOT / "data"
+    file = "first_run.txt"
+
+    if not os.path.isfile(data_path / file):
```

Comment:
the multiple `not`s in the conditional logic (starting with `if check and not check_first_run()` makes this hard to follow. thoughts on refactoring to not use the `not`s? 

maybe renaming the method would also help, eg `def first_run_started() -> bool` (my Ruby brain is looking for question-type methods for booleans). this would allow you to change the original conditional phrase  to be `if check and first_run_started()`, which IMO is much clearer.

Reply:
The file signals that a full run was completed, not that it has started. Maybe let's connect IRL to talk more about the logic?